components/patina_acpi/src/acpi.rs

```diff
@@ -113,8 +116,9 @@ where
     }
 
     /// Sets up tracking for the RSDP internally.
-    pub fn set_rsdp(&self, rsdp_table: AcpiTable) {
-        self.acpi_tables.write().insert(Self::RSDP_KEY, rsdp_table);
+    pub fn set_rsdp(&self, rsdp: &'static mut AcpiRsdp) {
```

Comment:
RwLock is prone to deadlock if it is actually used in an async context (like; a timer interrupt). I don't see that anywhere in this implementation, so probably fine; but since it's a public routine, might be worth a doc comment for callers to be aware that it is not safe to invoke simultaneously from multiple contexts. If invocation from a callback is a supported usage model, then probably need to switch to a TplMutex.